### PR TITLE
crate missing dependency when compiled on its own with std

### PR DESCRIPTION
**Pull request by gilescope**
This doesn't compile on its own as it needs `Debug` and that's
in the extra-traits syn feature.

### CODE REVIEW

File: frame/support/procedural/tools/Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,10 @@ description = "Proc macro helpers for procedural macros"
 [package.metadata.docs.rs]
 targets = ["x86_64-unknown-linux-gnu"]
 
+[features]
+default = [ "std" ]
+std = [ "syn/extra-traits" ]
+
```
Comment (bkchr) [inline frame/support/procedural/tools/Cargo.toml:17]:
```suggestion
```

Comment (bkchr) [reply frame/support/procedural/tools/Cargo.toml:17]: You need to enable the feature below for the syn dependency.

**Review by bkchr — CHANGES_REQUESTED** [review]

File: frame/support/procedural/tools/Cargo.toml (hunk 1 above)
Comment (gilescope) [reply frame/support/procedural/tools/Cargo.toml:17]: I can do that - I take it we're not too worried about compiling extra bits turned on by the extra-traits feature when compiling for no_std? That `extra-traits` feature is apparently compiler-heavy but I'm guessing that only applies to std... - it's certainly simpler to just add it in below as your suggesting.

Comment (bkchr) [reply frame/support/procedural/tools/Cargo.toml:17]: As you see below we already have extra features enabled. And this crate is never compiled for `no_std`. This is an extension for a derive macro and macros are always compiled for the host arch.

Comment (gilescope) [reply frame/support/procedural/tools/Cargo.toml:17]: "full" wasn't enough (at least on my machine from clean). 

"And this crate is never compiled for no_std." - maybe so, but it's the only crate of the 67 crates under frame that doesn't compile as std.

Comment (bkchr) [reply frame/support/procedural/tools/Cargo.toml:17]: I said that you should add the required feature below.



> "And this crate is never compiled for no_std." - maybe so, but it's the only crate of the 67 crates under frame that doesn't compile as std.

This crate never had a `std` feature and you added this.

Comment (gilescope) [reply frame/support/procedural/tools/Cargo.toml:17]: Ah you mean from a formatting perspective have the [features] below the [dependencies]. Sorry I misunderstood.

Comment (bkchr) [reply frame/support/procedural/tools/Cargo.toml:17]: No, I mean that you change:

```
syn = { version = "1.0.58", features = ["full", "visit"] }
```
to
```
syn = { version = "1.0.58", features = ["full", "visit", "extra-traits"] }
```

And nothing more.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
bot merge force

**Comment by bkchr** [comment]
bot merge force

**Pull request merged**